test/distributed/_composable/fsdp/test_fully_shard_memory.py

```diff
@@ -84,10 +84,10 @@ def _test_fully_shard_training_memory(
         mem_mb = self._get_peak_active_memory_mb()
         buffer_mb = 32  # 8.1 MiB cuBLAS workspaces, fragmentation, activations
         if reshard_after_forward:
-            # 2x max unsharded block parameters (all-gather + copy-out),
-            # non-block parameters, and other
+            # 3x max unsharded block parameters (current all-gather + copy-out
```

Comment:
In backward, the all-gather copy-out and the reduce-scatter can overlap. However, in forward, the copy-out is fully exposed and blocks the next copy-in.

We exploit the fact that the FSDP peak memory contribution happens in backward. In backward, we hold 3x layer size for current parameters, next all-gather, and current gradients, so we can similarly increase the memory usage in forward from 2x to 3x layer size.